lib/hpke/src/certificate.cpp

```diff
@@ -234,6 +234,34 @@ struct Certificate::ParsedCertificate
     return make_typed_unique<EVP_PKEY>(X509_get_pubkey(x509.get()));
   }
 
+  Certificate::Status status() const
+  {
+    auto* notBefore = X509_get_notBefore(x509.get());
+    auto ret = X509_cmp_current_time(notBefore);
+    if (ret == 0) {
```

Comment:
I'm not sure but `switch(X509_cmp_current_time(notBefore)) {...}` could potentially give better branch prediction.
Btw, did does it make sense to add attributes `[[likely]]`?